core/controllers/pages_test.py

```diff
@@ -15,14 +15,159 @@
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class ThanksPageTest(test_utils.GenericTestBase):
+    """Test for thanks page."""
+
+    def test_loading_thanks_page_loads_correctly(self):
+        response = self.testapp.get('/thanks')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class ForumPageTest(test_utils.GenericTestBase):
+    """Test for forum page."""
+
+    def test_loading_forum_page_loads_correctly(self):
+        response = self.testapp.get('/forum')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class TermsPageTest(test_utils.GenericTestBase):
+    """Test for terms page."""
+
+    def test_loading_terms_page_loads_correctly(self):
+        response = self.testapp.get('/terms')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class PrivacyPageTest(test_utils.GenericTestBase):
+    """Test for privacy page."""
+
+    def test_loading_privacy_page_loads_correctly(self):
+        response = self.testapp.get('/privacy')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class AboutRedirectPageTest(test_utils.GenericTestBase):
+    """Test for redirect to about page."""
+
+    def test_redirect_to_about_page_loads_correctly(self):
+        response = self.testapp.get('/credits')
+        self.assertEqual(response.status_int, 302)
+        self.assertIn('about', response.headers['location'])
+
+
+class FoundationRedirectPageTest(test_utils.GenericTestBase):
+    """Test for foundation redirect page."""
+
+    def test_redirect_to_foundation_page_loads_correctly(self):
+        response = self.testapp.get('/foundation')
+        self.assertEqual(response.status_int, 302)
+        self.assertIn('foundation', response.headers['location'])
+
+
+class TeachRedirectPageTest(test_utils.GenericTestBase):
+    """Test for teach redirect page."""
+
+    def test_redirect_to_teach_page_loads_correctly(self):
+        response = self.testapp.get('/participate')
+        self.assertEqual(response.status_int, 302)
+        self.assertIn('teach', response.headers['location'])
+
+
+class ConsoleErrorPageTest(test_utils.GenericTestBase):
+    """Test for console error page."""
+
+    def test_loading_console_error_page_loads_correctly(self):
+        response = self.testapp.get('/console_errors')
+        self.assertEqual(response.status_int, 200)
+
+
+class MaintenancePageTest(test_utils.GenericTestBase):
+    """Test for maintenance page."""
+
+    def test_loading_maintenance_page_loads_correctly(self):
+        with self.swap(feconf, 'ENABLE_MAINTENANCE_MODE', True):
+            response = self.testapp.get('/admin')
+            self.assertEqual(response.status_int, 302)
+            self.assertIn('admin', response.headers['location'])
```

Comment:
are we expecting to have a redirect to maintenance page?

Reply:
Hello @DubeySandeep , i was understand that when ENABLE_MAINTENANCE_MODE const is set true then url  redirect to admin page. I didn't find any page with maintenance url. can you guide me. Thank you.

Reply:
> are we expecting to have a redirect to maintenance page?

Hello @lilithxxx  could you address this query once, please. Thank you.

Reply:
> Hello @lilithxxx , i was understand that when ENABLE_MAINTENANCE_MODE const is set true then url redirect to admin page. But i didn't find any url for maintenance page. can you guide me. Thank you.

Hello @lilithxxx, I want to resolve this issue. so i request you to clarify my query once, please. Thank you.

Reply:
@netajik You can find it [here](https://github.com/oppia/oppia/blob/develop/main.py#L676)

Reply:
Done!